### PR TITLE
[JP install full plugin] Implement onboarding navigation

### LINKED ISSUE #17833: Create the fullscreen modal
Refs qk5s4q4mZsXknwhMWo2vIY-fi-8221%3A20788

Image

## Tasks

- [x] Create the fullscreen modal UI.
- [ ] Show the Jetpack plugin install flow when the Continue button is tapped.
- [x] Dismiss the modal and show My Sites when the Close button is tapped.
- [x] Show the Help screen when the Contact Support button is tapped.
- [x] Implement the presentation logic.

### PR DESCRIPTION
**Pull request by RenanLukas**
image
image

Fixes #17833 

To test:
This screen has no entry point yet. It will be testable when install full plugin "My Site" card is implemented.
To test the screen in this PR, you can use the following patch to add a test entry point:

```
diff --git a/WordPress/src/main/java/org/wordpress/android/ui/main/WPMainActivity.java b/WordPress/src/main/java/org/wordpress/android/ui/main/WPMainActivity.java
index 5903791640..23118c0b3f 100644
--- a/WordPress/src/main/java/org/wordpress/android/ui/main/WPMainActivity.java
+++ b/WordPress/src/main/java/org/wordpress/android/ui/main/WPMainActivity.java
@@ -91,6 +91,7 @@ import org.wordpress.android.ui.jetpackoverlay.JetpackFeatureFullScreenOverlayFr
 import org.wordpress.android.ui.jetpackoverlay.JetpackFeatureRemovalOverlayUtil;
 import org.wordpress.android.ui.jetpackoverlay.JetpackFeatureRemovalOverlayUtil.JetpackFeatureCollectionOverlaySource;
 import org.wordpress.android.ui.jetpackoverlay.JetpackFeatureRemovalPhaseHelper;
+import org.wordpress.android.ui.jpfullplugininstall.onboarding.JetpackFullPluginInstallOnboardingDialogFragment;
 import org.wordpress.android.ui.main.WPMainNavigationView.OnPageListener;
 import org.wordpress.android.ui.main.WPMainNavigationView.PageType;
 import org.wordpress.android.ui.mlp.ModalLayoutPickerFragment;
@@ -541,8 +542,10 @@ public class WPMainActivity extends LocaleAwareActivity implements
     }

     private void showBloggingPromptsOnboarding() {
-        BloggingPromptsOnboardingDialogFragment.newInstance(DialogType.ONBOARDING).show(
-                getSupportFragmentManager(), BloggingPromptsOnboardingDialogFragment.TAG
+//        BloggingPromptsOnboardingDialogFragment.newInstance(DialogType.ONBOARDING).show(
+//                getSupportFragmentManager(), BloggingPromptsOnboardingDialogFragment.TAG));
+        JetpackFullPluginInstallOnboardingDialogFragment.newInstance().show(
+                getSupportFragmentManager(), JetpackFullPluginInstallOnboardingDialogFragment.TAG
         );
     }
```
With the patch applied:
1 - Install Jetpack app;
2 - Select a site that is a blog;
3 - Tap the overflow menu in the prompt card;
4 - Select `"Learn more"`.

## Regression Notes
1. Potential unintended areas of impact
None

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing and unit testing.

3. What automated tests I added (or what prevented me from doing so)
Implemeted `JetpackFullPluginInstallOnboardingUiStateMapperTest.kt` and `JetpackFullPluginInstallOnboardingViewModelTest.kt`.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -4373,13 +4373,20 @@ translators: %s: Select control option value e.g: "Auto, 25%". -->
 
     <!-- Jetpack full plugin install onboarding -->
     <string name="jetpack_individual_plugin_support_onboarding_title">Please install the full Jetpack plugin</string>
-    <!-- translators: %1$s = the current site address (e.g. "https://wordpress.com/"), %2$s = the name of the plugin (e.g. "Jetpack Backup"), %3$s = the text "full Jetpack plugin" -->
-    <string name="jetpack_full_plugin_install_onboarding_description">%1$s is using the %2$s plugin, which doesn’t support all features of the app yet.\n\nPlease install the %3$s to use the app with this site.</string>
+    <!-- translators:
+        %1$s = the current site address (e.g. "https://wordpress.com/"),
+        %2$s = can be either "the <plugin name> plugin" (e.g. "the Jetpack Backup plugin") OR "individual plugins"
+        %3$s = the text "full Jetpack plugin" -->
+    <string name="jetpack_full_plugin_install_onboarding_description">%1$s is using %2$s, which don’t support all features of the app yet.\n\nPlease install the %3$s to use the app with this site.</string>
```
Comment (develric) [inline WordPress/src/main/res/values/strings.xml:4380]: np: not sure if it was discussed already, thinking of `which don't support...` should we consider singular vs plural? wdyt?

image

**Review by develric — APPROVED** [review]
Code looks good and works as per testplan 🙇 . I left a np, but approving this, since it's behind a feature flag, if my note makes sense feel free to address it on a subsequent PR as you prefer 🙇

File: WordPress/src/main/res/values/strings.xml (hunk 1 above)
Comment (RenanLukas) [reply WordPress/src/main/res/values/strings.xml:4380]: Thanks for catching that! As discussed in DM, it was decided to use `which don't support`.

**Comment by develric** [comment]
For traceability, we just found an issue when rotating the screen that was preventing the header and description field to show up.

| Before | After |
|---|---|
| image | image |

Thanks for fixing this in 30d6ad46ccdd8e0b92be7338ef37ec9fff323b70 @RenanLukas 🙇 ...I'm going ahead and merging this one 👍

**Pull request merged**